### PR TITLE
[Enhancement] Rebuild persistent index if load previous index failed

### LINKED ISSUE #8141: Rebuild persistent index when some unexpect error happened
## Enhancement
If a primary key table using persistent index meets some error when try to build persistent index from l0 file, the table state will be set to error and disable read or write. Maybe we can rebuild the persistent index instead of return false.

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/starrocks/issues/8141 https://github.com/StarRocks/starrocks/issues/10246

## Problem Summary(Required) ：

We will try to build a persistent index according to the index file.  If some error occurs during the loading index, we will return fail directly. But we can rebuild the persistent index according to tablet data. So we will try to rebuild the persistent index when some error occurs.

In addition, we will support varlen key for the PrimaryKey table with a persistent index from version 2.4. However, if we create a PrimaryKey table with a persistent index whose key length is less than 64 Bytes in version 2.4, we can't rollback to version 2.3 directly because the meta structure is changed in version 2.4.

This pr also supports rollback from version 2.4 to version 2.3 for PrimaryKey table with a persistent index.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by sevev** [comment]
run starrocks_admit_test

File: be/src/storage/persistent_index.cpp [hunk 1]
```diff
@@ -1569,14 +1569,39 @@ Status PersistentIndex::load_from_tablet(Tablet* tablet) {
...
-                      << " memory: " << memory_usage() << " status: " << status.to_string()
-                      << " time:" << timer.elapsed_time() / 1000000 << "ms";
-            return status;
+            MutableIndexMetaPB l0_meta = index_meta.l0_meta();
+            if (l0_meta.format_version() != PERSISTENT_INDEX_VERSION_1) {
+                LOG(WARNING) << "different format version, we need to rebuild persistent index";
+                status = Status::InternalError("different format version");
+            } else {
+                status = load(index_meta);
+                LOG(INFO) << "load persistent index tablet:" << tablet->tablet_id()
```
Comment (decster) [inline be/src/storage/persistent_index.cpp:1578]: should only print this when status.ok()

**Review by decster — DISMISSED** [review]

File: be/src/storage/persistent_index.cpp (hunk 1 above)
Comment (sevev) [reply be/src/storage/persistent_index.cpp:1580]: > should only print this when status.ok()

done, change inhttps://github.com/StarRocks/starrocks/pull/10244/commits/1753328b958be5c46afa3c3869c2617bd6450c99

**Comment by sevev** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
run starrocks_admit_test

**Comment by sevev** [comment]
@decster PTAL

**Review by decster — APPROVED** [review]

**Comment by sevev** [comment]
@mergify backport branch-2.3

**Pull request merged**